src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash3.cs

```diff
@@ -896,12 +896,24 @@ private static Vector128<ulong> Accumulate128(Vector128<ulong> accVec, byte* sou
             Vector128<uint> sourceKey = sourceVec ^ secret;
 
             // TODO: Figure out how to unwind this shuffle and just use Vector128.Multiply
-            Vector128<uint> sourceKeyLow = Vector128.Shuffle(sourceKey, Vector128.Create(1u, 0, 3, 0));
             Vector128<uint> sourceSwap = Vector128.Shuffle(sourceVec, Vector128.Create(2u, 3, 0, 1));
             Vector128<ulong> sum = accVec + sourceSwap.AsUInt64();
-            Vector128<ulong> product = Sse2.IsSupported ?
-                Sse2.Multiply(sourceKey, sourceKeyLow) :
-                (sourceKey & Vector128.Create(~0u, 0u, ~0u, 0u)).AsUInt64() * (sourceKeyLow & Vector128.Create(~0u, 0u, ~0u, 0u)).AsUInt64();
+
+            Vector128<ulong> product;
+            // TODO: Workaround for ARM as `*` is software emulated.
+            if (System.Runtime.Intrinsics.Arm.AdvSimd.IsSupported)
```

Comment:
In other places for a code written with cross-plat helpers we hide such routines under helpers, like
```csharp
static void MultiplyLongs(Vector128<ulong> v1, Vector128<ulong> v2)
{
    if (AdvSimd.IsSupported)
    {
    }
    else if (Sse2.IsSupported)
    {
    }
    else
    {
        // fallback
    }
}
```
for better readability of the general algorithm

Reply:
Hopefully fixed by commit 6b07c09